backend-tests/tests/test_monitoring.py

```diff
@@ -0,0 +1,205 @@
+
+import pytest
+import uuid
+import time
+
+from datetime import datetime
+
+from testutils.api import useradm, devicemonitor, deviceauth
+from testutils.api.client import ApiClient
+from testutils.common import (
+    clean_mongo,
+    create_authset,
+    create_org,
+    create_random_authset,
+    create_user,
+    make_accepted_device,
+    make_pending_device,
+    mongo,
+)
+from testutils.infra.cli import CliUseradm, CliDeviceauth, CliDeviceMonitor
+from testutils.infra.container_manager.kubernetes_manager import isK8S
+from testutils.infra.smtpd_mock import smtp_mock
+
+
+@pytest.fixture(scope="function")
+def clean_migrated_mongo(clean_mongo):
+    CliDeviceauth().migrate()
+    CliUseradm().migrate()
+    CliDeviceMonitor().migrate()
+    yield clean_mongo
+
+
+@pytest.fixture(scope="function")
+def tenants_users(clean_migrated_mongo):
+    tenants = []
+    for n in range(2):
+        uuidv4 = str(uuid.uuid4())
+        tenant, username, password = (
+            "test.mender.io-" + uuidv4,
+            "some.user+" + uuidv4 + "@example.com",
+            "secretsecret",
+        )
+        tenants.append(create_org(tenant, username, password))
+
+    yield tenants
+
+
+@pytest.fixture(scope="function")
+def user(clean_migrated_mongo):
+    yield create_user("user-foo@acme.com", "correcthorse")
+
+
+@pytest.fixture(scope="function")
+def tenants_users_devices(tenants_users):
+    uc = ApiClient(useradm.URL_MGMT)
+    devauthm = ApiClient(deviceauth.URL_MGMT)
+    devauthd = ApiClient(deviceauth.URL_DEVICES)
+
+    for t in tenants_users:
+        user = t.users[0]
+        r = uc.call("POST", useradm.URL_LOGIN, auth=(user.name, user.pwd))
+        assert r.status_code == 200
+        utoken = r.text
+
+        for _ in range(5):
+            dev = make_accepted_device(devauthd, devauthm, utoken, t.tenant_token)
+            t.devices.append(dev)
+
+    yield tenants_users
+
+
+@pytest.fixture(scope="function")
+def devices(user):
+    uc = ApiClient(useradm.URL_MGMT)
+    devauthm = ApiClient(deviceauth.URL_MGMT)
+    devauthd = ApiClient(deviceauth.URL_DEVICES)
+
+    r = uc.call("POST", useradm.URL_LOGIN, auth=(user.name, user.pwd))
+    assert r.status_code == 200
+    utoken = r.text
+
+    devices = []
+
+    for _ in range(5):
+        dev = make_accepted_device(devauthd, devauthm, utoken)
+        devices.append(dev)
+
+    yield devices
+
+
+class _TestMonitoringAlertsBase:
+    useradm = ApiClient(useradm.URL_MGMT)
+    devmond = ApiClient(devicemonitor.URL_DEVICES)
+
+    def test_alerting_email(self, test_case, user, devices, smtp_mock):
+        device = devices.pop(1)
+        for alert in test_case["alerts"]:
+            r = self.devmond.with_auth(device.token).call(
+                "POST", devicemonitor.URL_ALERT, alert,
+            )
+            assert r.status_code < 300
+
+        try:
+            wait_start = datetime.now()
+            smtp_mock.await_messages(len(test_case["alerts"]), 60.0)
```

Comment:
do you think it would betoo much to at least verify that the Subject: field contains OK and CRITICAL strings?